### PR TITLE
Feat/elia 146 local cache server

### PR DESCRIPTION
**Pull request by jrhender**
**Review this PR after PR #15 as it has the changes from that PR**

The crux of this change is that iam-client-factory now accepts a `cacheServerUrl` and config has two separate cacheServerUrls, one for "users" and one for "roles". This is because the "prod" cacheserver has the user roles while local cache-server allows for nats. We could, in theory, set up user roles in dev as well, but it's nice to not have to do that.

In addition, I think the logical separation of users and assets as maybe one day there will be separate mechanisms for user claims and asset claims.

### CODE REVIEW

File: src/factories/nats-connection-factory.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+/**
+ * Creates and monitors a NATS connection
+ * Event handling code taken from: https://docs.nats.io/developing-with-nats/events/events
+ */
+export class NatsConnectionFactory {
+    static readonly logPrefix = "[NATS Event]";
+
+    static create(): Client | undefined {
+        try {
+            const assetIamConfig = config.prequalification.asset_claims_iam;
```
Comment (adamstaveley) [inline src/factories/nats-connection-factory.ts:13]: minor thing but inconsistent case

**Review by adamstaveley — APPROVED** [review]
Looks as intended.

File: src/factories/nats-connection-factory.ts (hunk 1 above)
Comment (jrhender) [reply src/factories/nats-connection-factory.ts:13]: Good point

Comment (jrhender) [reply src/factories/nats-connection-factory.ts:13]: Actually, sorry, I'm not quite following, did you mean that `assetIamConfig` casing is inconsistent?

**Pull request merged**